src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -376,13 +376,55 @@ internal void FunctionEnvironmentReloadResponse(FunctionEnvironmentReloadRespons
 
             if (res.Result.IsFailure(out Exception reloadEnvironmentVariablesException))
             {
-                _workerChannelLogger.LogError(reloadEnvironmentVariablesException, "Failed to reload environment variables");
-                _reloadTask.SetException(reloadEnvironmentVariablesException);
+                if (IsEnvironmentReloadFailureFatalError(res))
```

Comment:
What are the cases where it's not fatal? I assumed that if the worker says it failed, we assume it's fatal. 

No one else even uses this result (b/c it never worked), so we can define the scenario...